component/config.go

```diff
@@ -164,3 +164,15 @@ var (
 	// DataTypeLogs is the data type tag for logs.
 	DataTypeLogs = mustNewDataType("logs")
 )
+
+// nameRegexp is used to validate the name of a component.
+// A name must start with a Unicode alphanumeric character and
+// can only contain Unicode alphanumeric characters and punctuation.
+var nameRegexp = regexp.MustCompile(`^[\pN\pL][\pN\pL\pP]{0,62}$`)
```

Comment:
Should we just follow https://www.unicode.org/reports/tr31/#Default_Identifier_Syntax?

Reply:
I interpreted this as use the long form for category name. Let me know if that was what you had in mind or if it was something else.

Reply:
I took @mx-psi's comment to mean following the syntax `<Identifier> := <Start> <Continue>* (<Medial> <Continue>+)*`, where `<Start>`, `<Continue>`, and `<Medial>` are defined in that page in terms of various Unicode categories.

I like the idea of using a standard, but OTOH this one seems quite complicated. There's also the relatively simple https://www.unicode.org/reports/tr31/#Immutable_Identifier_Syntax. Essentially: disallow spaces, syntax, control characters. That might be good enough here?

If it's not straightforward, maybe we could start with something something strict (e.g. a subset of both of the above) and open an issue to expand the allowed names?

Reply:
> I took @mx-psi's comment to mean following the syntax <Identifier> := <Start> <Continue>* (<Medial> <Continue>+)*, where <Start>, <Continue>, and <Medial> are defined in that page in terms of various Unicode categories.

Yes, that was it

>  Essentially: disallow spaces, syntax, control characters. That might be good enough here?

This sounds reasonable as a start

> a subset of both of the above

That also sounds good. On `component.ID` we did discuss Unicode support  and ended up restricting to ASCII-only for now. I think here there are other considerations here since the user should have full control over the name so
- it's more likely we break existing users if we restrict this to ASCII only
- users can stop using non-ASCII characters if needed (e.g. if sending telemetry related to the Collector itself to a backend that does not have full Unicode support)

Reply:
I took another look at the default identifier. In addition to its complexity, it would prohibit a name from starting with a digit. This is something we have in our test cases (e.g. `traces/1`). I'm not sure how much it would be used in the wild. I took the least restrictive approach, which is to restrict the name to 1 to 63 Unicode characters excluding whitespace, control characters, and symbols.

Reply:
Makes sense, thanks Matt!